tools/internal_ci/helper_scripts/prepare_build_macos_rc

```diff
@@ -48,6 +48,10 @@ set -ex
 
 # cocoapods
 export LANG=en_US.UTF-8
+# pre-fetch cocoapods master repo with HEAD only
```

Comment:
I don't quite understand the remark that it fetches with "HEAD" only - it's not clear from the
command line that this is what's being done.
I'm not an expert, but it sounds like you're wanting to do `--single-branch` to only fetch the master branch and you actually might be doing that, as --depth seems to imply `--single-branch` but without further investigation, that's probably not going to be obvious to someone who looks at this script.

Reply:
> but it sounds like you're wanting to do `--single-branch` to only fetch the master branch and you actually might be doing that

`-depth 1` only fetches one commit in the history tree. According to the [doc](https://git-scm.com/docs/git-clone), it indeed implies `--single-branch`, but that does not matter much in this case because the cocoapods master repo only has a single branch AFAIK.

Reply:
Ok, but perhaps the comment could be clearer. I'd like to also see a comment that explains how the particular type of clone we do later interacts with whatever happens in `pod install` (which also seems to clone/update under some circumstances) and how that affect the total runtime of the tests.

Reply:
> We'll see what the new patch does

I'm looking at the new test results and I think the timings have improved, but I think we need to understand this more, because right now the behavior doesn't make much sense to me.